### PR TITLE
M-134

### PR DESCRIPTION
**Pull request by Luksys3**
https://trello.com/c/n1qs258O/134-add-deadlines-and-timestamps

### CODE REVIEW

File: database/factories/RoomFactory.php [hunk 1]
```diff
@@ -43,6 +44,15 @@ public function configure(): RoomFactory
...
 	{
+		$startDateTime = $this->faker->boolean
+			? CarbonImmutable::now()->addDays(
+				$this->faker->numberBetween(-6, 5)
+			)
+			: null;
+		$endDateTime = $startDateTime?->addDays(
+			$this->faker->numberBetween(1, 6)
+		);
+
```
Comment (gustass5) [inline database/factories/RoomFactory.php:55]: With this seed, is it possible for start date to be in front of end date?

File: domain/API/Controllers/GetRoomWithAccessCodeController.php [hunk 2]
```diff
@@ -4,19 +4,34 @@
...
 
 class GetRoomWithAccessCodeController
 {
 	public function __invoke(Room $room): JsonResponse
 	{
+		$now = CarbonImmutable::now();
+
+		if ($room->started_at !== null && $now->isBefore($room->started_at)) {
+			return ApiJsonResponse::make(403)
+				->setError("Room not started")
```
Comment (gustass5) [inline domain/API/Controllers/GetRoomWithAccessCodeController.php]: Perhaps change it into something dynamic like `Room is unavailable until ${start_date}`

File: domain/API/Controllers/GetRoomWithAccessCodeController.php [hunk 3]
```diff
@@ -4,19 +4,34 @@
...
+				->export();
+		}
+
+		if ($room->ended_at !== null && $now->isAfter($room->ended_at)) {
+			return ApiJsonResponse::make(403)
+				->setError("Room ended")
```
Comment (gustass5) [inline domain/API/Controllers/GetRoomWithAccessCodeController.php]: `Room has ended`, but I would prefer `Room is no longer available`

File: database/factories/RoomFactory.php (hunk 1 above)
Comment (Luksys3) [reply database/factories/RoomFactory.php:55]: start date will always be before end date

**Review by gustass5 — APPROVED** [review]

**Pull request merged**